typing/typedtree.mli

```diff
@@ -35,8 +35,10 @@ type attributes = attribute list
 
 (** {1 Core language} *)
 
-type pattern =
-  { pat_desc: pattern_desc;
+type pattern = pattern_desc pattern_
+
+and 'a pattern_ =
```

Comment:
`pattern_node` ?
Please reserve the final `_` for things like keywords, it looks like something you didn't want to name.

Reply:
There are situations (not in this PR, but in another forthcoming PR!) where we want to manipulate several distinct instances of `'a pattern_` in the same codebase, to distinguish various sub-classes of patterns in the pattern-matching analysis and compilation code. So it really helps if the name we choose for this one is short enough -- `pattern_node` would give type signatures that are really verbose. Would you prefer `'a pat` to `'a pattern_`? (Personally I like `pattern_` more, but I would be fine with `pat`).

Note: the parser has used `foo_` to mean "the auxiliary variant of `foo`" for a while now, so there is an established usage of this naming scheme outside keywords. But I don't mind using something else as long as it remains reasonably compact.

Reply:
`pat` or `pattern'` are ok. For me something ending with `_` is not a name.